binaries/aot_model_compiler.cc

```diff
@@ -61,125 +54,20 @@ std::vector<std::string> split(
   return pieces;
 }
 
-std::vector<std::vector<int64_t>> parseInputShapes() {
-  CAFFE_ENFORCE_GE(FLAGS_input_dims.size(), 0, "Input dims must be specified.");
-  std::vector<std::string> input_dims_list = split(';', FLAGS_input_dims);
-  std::vector<std::vector<int64_t>> inputs;
-  for (const auto& input_dims_item : input_dims_list) {
-    auto input_dims_str = split(',', input_dims_item);
-    std::vector<int64_t> input_dims;
-    input_dims.reserve(input_dims_str.size());
-    for (const auto& s : input_dims_str) {
-      input_dims.push_back(c10::stoi(s));
-    }
-    inputs.push_back(input_dims);
-  }
-  return inputs;
-}
-
-std::vector<at::ScalarType> parseInputTypes() {
-  std::vector<std::string> inputTypes = split(';', FLAGS_input_types);
-  std::vector<at::ScalarType> scalarTypes;
-  for (const auto& inputType : inputTypes) {
-    at::ScalarType scalarType;
-    if (inputType == "float") {
-      scalarType = at::ScalarType::Float;
-    } else if (inputType == "uint8") {
-      scalarType = at::ScalarType::Byte;
-    } else if (inputType == "int64") {
-      scalarType = at::ScalarType::Long;
-    } else {
-      CAFFE_THROW("Unsupported input type: ", inputType);
-    }
-    scalarTypes.push_back(scalarType);
-  }
-  return scalarTypes;
-}
-
 c10::Dict<c10::IValue, c10::IValue> createCompileSpec() {
   c10::Dict<c10::IValue, c10::IValue> compile_spec(
       c10::StringType::get(), c10::AnyType::get());
   c10::Dict<c10::IValue, c10::IValue> method_spec(
       c10::StringType::get(), c10::AnyType::get());
-  auto inputShapes = parseInputShapes();
-  auto inputTypes = parseInputTypes();
-  method_spec.insert("sizes", inputShapes);
-  method_spec.insert("types", inputTypes);
+  method_spec.insert("sizes", FLAGS_input_dims);
+  method_spec.insert("types", FLAGS_input_types);
+  method_spec.insert("asmfile", FLAGS_output_llvm);
+  method_spec.insert("model_name", FLAGS_model_name);
```

Comment:
model_name and model_version can be in the compile_spec outside the method_spec ?

Reply:
There is a bunch of places where we iterate over compile spec and expect keys to be method names, so we can't just put model_name and model_version there. Would you mind if I clean that up in a separate PR after this lands?

Reply:
Ah that's right there are places where we expect keys to be method names. I'll look at what other delegates are doing and converge them.